tensorflow_model_optimization/python/core/sparsity_tf2/pruner_test.py

```diff
@@ -0,0 +1,493 @@
+  #   expected_mask = [[0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0],
+  #                    [1., 1., 1., 1.], [1., 1., 1., 1.]]
+
+  #   self._blockMasking(block_size, block_pooling_type, weight, expected_mask)
+
+  # def testBlockMaskingWithHigherDimensionsRaisesError(self):
+  #   self.initialize()
+  #   block_size = (2, 2)
+  #   block_pooling_type = "AVG"
+  #   # Weights as in testBlockMasking, but with one extra dimension.
+  #   weight = tf.constant([[[0.1, 0.1, 0.2, 0.2], [0.1, 0.1, 0.2, 0.2],
+  #                                   [0.3, 0.3, 0.4, 0.4], [0.3, 0.3, 0.4,
+  #                                                          0.4]]])
+  #   expected_mask = [[[0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0],
+  #                     [1., 1., 1., 1.], [1., 1., 1., 1.]]]
+
+  #   # Block masking should only be used with 2 Dimensional weights.
+  #   with self.assertRaises(ValueError):
+  #     self._blockMasking(block_size, block_pooling_type, weight, expected_mask)
+
+  # def testConditionalMaskUpdate(self):
+  #   weight = tf.Variable(np.linspace(1.0, 100.0, 100), name="weights")
+  #   weight_dtype = weight.dtype.base_dtype
+  #   mask = tf.Variable(
+  #       tf.ones(weight.get_shape(), dtype=weight_dtype),
+  #       name="mask",
+  #       dtype=weight_dtype)
+  #   threshold = tf.Variable(
+  #       tf.zeros([], dtype=weight_dtype), name="threshold", dtype=weight_dtype)
+  #   self.initialize()
+  #   pruning_vars = [(weight, mask, threshold)]
+
+  #   def linear_sparsity(step):
+  #     sparsity_val = tf.convert_to_tensor(
+  #         [0.0, 0.1, 0.1, 0.3, 0.3, 0.5, 0.5, 0.5, 0.5, 0.5])
+  #     return tf.convert_to_tensor(True), sparsity_val[step]
+
+  #   def weight_mask_op(pruning_vars):
+  #     values_and_vars = []
+  #     for weight, mask, _ in  pruning_vars:
+  #       # values_and_vars.append((tf.math.multiply(weight, mask), weight))
+  #       weight.assign(tf.math.multiply(weight, mask))
+  #     # return tf.group(values_and_vars)
+
+  #   # Set up pruning
+  #   p = pruner.LTHPruner(
+  #       pruning_schedule=linear_sparsity,
+  #       save_schedule=self.save_init,
+  #       block_size=self.block_size,
+  #       block_pooling_type=self.block_pooling_type)
+    
+  #   step = self.training_step_fn
+
+  #   non_zero_count = []
+  #   for _ in range(10):
+  #     if tf.executing_eagerly():
+  #       p.update_masks(pruning_vars, step())
+  #       weight_mask_op(pruning_vars)
+  #       assign_add(self.global_step, 1)
+  #     else:
+  #       K.get_session().run(p.update_masks(pruning_vars, step()))
+  #       K.get_session().run(weight_mask_op(pruning_vars))
+  #       K.get_session().run(assign_add(self.global_step, 1))
+
+  #     non_zero_count.append(np.count_nonzero(K.get_value(weight)))
+
+  #   # Weights pruned at steps 1,3,5
+  #   expected_non_zero_count = [100, 90, 90, 70, 70, 50, 50, 50, 50, 50]
+  #   self.assertAllEqual(expected_non_zero_count, non_zero_count)
+
+
+  def testSaveOGSlotInit(self):
```

Comment:
`testSaveOriginalInitializations`

Only abbreviate if it's unambiguous & a common abbreviation.